ci/gpu/build.sh

```diff
@@ -12,7 +12,7 @@ function logger() {
 
 # Set path and build parallel level
 export PATH=/conda/bin:/usr/local/cuda/bin:$PATH
-export CMAKE_BUILD_PARALLEL_LEVEL=4
+export PARALLEL_LEVEL=4
```

Comment:
Just a note this won't pass to the Arrow / GTest builds. The `CMAKE_BUILD_PARALLEL_LEVEL` would pass to it if we didn't set the `PARALLEL_LEVEL` CMake variable.

Reply:
Right I use this below for the manual build process when I call `make` further down in the script